components/VM_Arm/src/main.c

```diff
@@ -831,32 +832,48 @@ static int load_generated_dtb(vm_t *vm, uintptr_t paddr, void *addr, size_t size
     return 0;
 }
 
-static int load_linux(vm_t *vm, const char *kernel_name, const char *dtb_name, const char *initrd_name)
+static int load_vm(vm_t *vm, const char *kernel_name, const char *dtb_name, const char *initrd_name)
 {
     seL4_Word entry;
     seL4_Word dtb;
     int err;
 
+    vm->mem.map_one_to_one = vm_image_config.map_one_to_one; /* Map memory 1:1 if configured to do so */
+
     /* Install devices */
-    err = install_linux_devices(vm);
+    err = install_vm_devices(vm);
     if (err) {
         printf("Error: Failed to install Linux devices\n");
         return -1;
     }
 
+    /* If the entry_addr isn't set, just use the previous default logic */
+    if (!entry_addr) {
```

Comment:
- We should print a warning here to really encourage users to set this. It would be nice of we could remove any platform code here, so this must be parameter that i specified explicitly. That would make things more transparent. 
- Since we have guest OS agnostic now, can we be sure `entry_addr` can't be `0` also? This might be another argument to remove any smart default here that may cause strange side effects. It may break compatibility, but I wonder if this is a price to pay for a more well defined behavior without internal magic.

Reply:
I've added a ZF_LOGW if entry_addr is not set.

For your second point, it would technically be possible to have RTEMS boot at 0, but for Linux, there's always an offset.